app/models/course/condition/achievement.rb

```diff
@@ -32,7 +32,7 @@ def self.dependent_class
   end
 
   def self.on_dependent_status_change(achievement)
-    return if achievement.previous_changes.empty?
+    return if achievement.changes.empty?
```

Comment:
I guess the specs should be fixed to catch this. I don't think the current changes to specs would catch this.